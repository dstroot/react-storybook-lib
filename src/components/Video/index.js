import React from "react";
// import poster from "../img/video-placeholder.png";

// https://www.sample-videos.com/
// https://developers.google.com/web/fundamentals/media/video

const style = {
  width: "100vw",
  height: "auto"
};

const Video = ({ videoURL, poster }) => (
  <video style={style} preload="auto" poster={poster} loop autoPlay muted>
    {/* List formats in order of preference. */}
    <source src={videoURL} type="video/webm" />
    <source src={videoURL} type="video/mp4" />

    {/* older browser */}
    <p>This browser does not support the video element.</p>
  </video>
);

export default Video;

// <Video videoURL="https://www.sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4" />
