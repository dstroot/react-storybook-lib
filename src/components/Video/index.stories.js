import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";

// components
import Video from "../Video";

// Video section
storiesOf("Video", module)
  .addDecorator(withKnobs)
  .addParameters({
    info: {
      inline: true,
      header: false
    }
  })
  .add("show Video", () => (
    <Video
      videoURL={text(
        "Video URL",
        "//videos.ctfassets.net/uw229zq1tzc4/3CYVQZf6e4EyACIEEKOaio/867dda9b5beb1b7d8f1c34261029b02a/HiRes_CC_Text_Removed.mp4"
      )}
      poster={text(
        "Poster",
        "//images.ctfassets.net/uw229zq1tzc4/6mhWavUAFUGrfKwkylCPqM/c2dcbb66c2959c1fe8164cf779d3302b/home-hero-poster-image.jpg"
      )}
    />
  ));

//videos.ctfassets.net/uw229zq1tzc4/3CYVQZf6e4EyACIEEKOaio/867dda9b5beb1b7d8f1c34261029b02a/HiRes_CC_Text_Removed.mp4
//www.sample-videos.com/video123/mp4/720/big_buck_bunny_720p_2mb.mp4
