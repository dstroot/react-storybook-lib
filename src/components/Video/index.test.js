import React from "react";
import renderer from "react-test-renderer";
import Video from "../Video";

describe("Video", () => {
  it("it should render", () => {
    const component = renderer.create(
      <Video
        videoURL="//videos.ctfassets.net/uw229zq1tzc4/3CYVQZf6e4EyACIEEKOaio/867dda9b5beb1b7d8f1c34261029b02a/HiRes_CC_Text_Removed.mp4"
        poster="//images.ctfassets.net/uw229zq1tzc4/6mhWavUAFUGrfKwkylCPqM/c2dcbb66c2959c1fe8164cf779d3302b/home-hero-poster-image.jpg"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
