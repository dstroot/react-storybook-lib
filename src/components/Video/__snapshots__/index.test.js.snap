// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video it should render 1`] = `
<video
  autoPlay={true}
  loop={true}
  muted={true}
  poster="//images.ctfassets.net/uw229zq1tzc4/6mhWavUAFUGrfKwkylCPqM/c2dcbb66c2959c1fe8164cf779d3302b/home-hero-poster-image.jpg"
  preload="auto"
  style={
    Object {
      "height": "auto",
      "width": "100vw",
    }
  }
>
  <source
    src="//videos.ctfassets.net/uw229zq1tzc4/3CYVQZf6e4EyACIEEKOaio/867dda9b5beb1b7d8f1c34261029b02a/HiRes_CC_Text_Removed.mp4"
    type="video/webm"
  />
  <source
    src="//videos.ctfassets.net/uw229zq1tzc4/3CYVQZf6e4EyACIEEKOaio/867dda9b5beb1b7d8f1c34261029b02a/HiRes_CC_Text_Removed.mp4"
    type="video/mp4"
  />
  <p>
    This browser does not support the video element.
  </p>
</video>
`;
